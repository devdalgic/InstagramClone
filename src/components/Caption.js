import React from 'react';
import { StyleSheet, Text, View } from 'react-native';

export const Caption = ({ postedBy, caption }) => {
  return (
    <View style={styles.captionContainer}>
      <Text style={styles.captionText}>
        <Text style={styles.captionUsernameText}> {postedBy} </Text>
        {caption}{' '}
      </Text>
    </View>
  );
};

const styles = StyleSheet.create({
  captionContainer: {
    flex: 1,
    flexDirection: 'row',
    marginStart: 10,
  },
  captionUsernameText: {
    fontWeight: 'bold',
  },
  captionText: {
    color: 'black',
    flexShrink: 1,
  },
});
