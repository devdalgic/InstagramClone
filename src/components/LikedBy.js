import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import PropTypes from 'prop-types';

/**
 * Component to show who liked a post along with others.
 */
export const LikedBy = ({ likedBy }) => {
  return (
    <View style={styles.likedByContainer}>
      <Text style={styles.text}>
        Liked by
        <Text style={styles.usernameText}> {likedBy} </Text>
        and others{' '}
      </Text>
    </View>
  );
};

LikedBy.propTypes = {
  likedBy: PropTypes.string.isRequired,
};

const styles = StyleSheet.create({
  likedByContainer: {
    flex: 1,
    flexDirection: 'row',
    marginTop: 10,
  },
  usernameText: {
    fontWeight: 'bold',
  },
  text: {
    color: 'black',
  },
});
