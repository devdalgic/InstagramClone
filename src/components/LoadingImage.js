import React, { useState } from 'react';
import { Image, View, ActivityIndicator, StyleSheet } from 'react-native';

export const LoadingImage = props => {
  const [isLoading, setIsLoading] = useState(true);
  const onLoadEnd = () => {
    setIsLoading(false);
  }
  return (
    <View style={styles.container}>
      <Image
        source={{ uri: props.source }}
        style={styles.image}
        onLoadEnd={onLoadEnd}
      />
      <ActivityIndicator
        style={styles.activityIndicator}
        color={'#841584'}
        animating={isLoading}
        size={'large'}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  image: {
    width: 500,
    height: 300,
  },
  activityIndicator: {
    position: 'absolute',
    left: 0,
    right: 0,
    top: 0,
    bottom: 0,
  },
});
