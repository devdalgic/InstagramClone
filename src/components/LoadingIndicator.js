import React from 'react';
import { ActivityIndicator, StyleSheet } from 'react-native';
import PropTypes from 'prop-types';

/**
 * An Activity Indicator with some predefined values
 */
export const LoadingIndicator = ({ animating = true, color = '#841584' }) => {
  return (
    <ActivityIndicator
      animating={animating}
      size={'large'}
      color={color}
      style={styles.activityIndicator}
    />
  );
};

LoadingIndicator.propTypes = {
  animating: PropTypes.bool,
  color: PropTypes.string,
};

const styles = StyleSheet.create({
  activityIndicator: {
    position: 'absolute',
    left: 0,
    right: 0,
    top: 0,
    bottom: 0,
  },
});
