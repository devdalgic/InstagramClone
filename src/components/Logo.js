import React from 'react';
import Icon from 'react-native-vector-icons/FontAwesome5';
import { StyleSheet, Text, View } from 'react-native';

export const Logo = () => {
  return (
    <View style={styles.logoContainer}>
      <Icon name="cat" color={'#841584'} size={192} style={styles.logo} />
      <Text style={styles.info}>Case Study by Denizhan Dalgic</Text>
      <Text style={styles.info}>Instagram Clone</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  logoContainer: {
    alignItems: 'center',
  },
  logo: {
    marginVertical: 10,
  },
  info: {
    color: 'black',
  },
});
