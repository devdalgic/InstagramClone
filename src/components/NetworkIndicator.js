import React, { useEffect, useState } from 'react';
import NetInfo from '@react-native-community/netinfo';
import { StyleSheet, Text, View } from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';

export const NetworkIndicator = () => {
  const [isConnected, setIsConnected] = useState(true);

  useEffect(() => {
    NetInfo.addEventListener(state => {
      setIsConnected(state.isConnected);
    });
  }, []);

  return isConnected ? (
    <View />
  ) : (
    <View style={styles.container}>
      <View style={styles.textContainer}>
        <Icon name={'warning'} color={'red'} style={styles.icon} size={24} />
        <Text style={styles.text}>Network Error</Text>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    position: 'absolute',
    left: 10,
    right: 10,
    bottom: 10,
    flex: 1,
    flexDirection: 'row',
    backgroundColor: 'white',
    borderRadius: 5,
  },
  textContainer: {
    margin: 10,
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
  },
  icon: {
    marginEnd: 10,
  },
  text: {
    color: 'black',
  },
});
