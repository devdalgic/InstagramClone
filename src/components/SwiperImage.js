import React from 'react';
import { FlatList, StyleSheet } from 'react-native';
import { width } from '../utils/constants';
import { LoadingImage } from './LoadingImage';

export const SwiperImage = ({ media }) => {
  const renderItem = item => {
    return <LoadingImage source={item} style={styles.image} />;
  };

  return (
    <FlatList
      data={media}
      renderItem={({ item }) => renderItem(item)}
      horizontal={true}
      decelerationRate={'normal'}
      snapToInterval={width}
      snapToAlignment={'start'}
      showsHorizontalScrollIndicator={false}
    />
  );
};

const styles = StyleSheet.create({
  image: {
    width: 500,
    height: 300,
  },
});
