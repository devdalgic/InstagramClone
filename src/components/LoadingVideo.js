import React, { useState } from 'react';
import Video from 'react-native-video';
import { LoadingIndicator } from './LoadingIndicator';

export const LoadingVideo = ({ source, style }) => {
  const [isLoading, setIsLoading] = useState(true);
  return (
    <>
      <Video
        source={{ uri: source }}
        style={style}
        repeat={true}
        posterResizeMode={'cover'}
        resizeMode={'cover'}
        onBuffer={({ isBuffering }) => setIsLoading(isBuffering)}
      />
      <LoadingIndicator animating={isLoading} />
    </>
  );
};
