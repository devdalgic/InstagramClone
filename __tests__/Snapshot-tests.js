import React from 'react';
import renderer from 'react-test-renderer';
import { Caption } from '../src/components/Caption';
import { LoadingVideo } from '../src/components/LoadingVideo';
import { NetworkIndicator } from '../src/components/NetworkIndicator';
import { LikedBy } from '../src/components/LikedBy';
import { LoadingImage } from '../src/components/LoadingImage';
import { LoadingIndicator } from '../src/components/LoadingIndicator';
import { Logo } from '../src/components/Logo';
import { SwiperImage } from '../src/components/SwiperImage';
import { UserHeader } from '../src/components/UserHeader';

/**
 * Snapshot tests to test component trees.
 */

test('Caption renders correctly', () => {
  const tree = renderer
    .create(<Caption postedBy={'test'} caption={['test']} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('LikedBy renders correctly', () => {
  const tree = renderer.create(<LikedBy likedBy={'test'} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('LoadingImage renders correctly', () => {
  const tree = renderer
    .create(<LoadingImage source={'test'} style={{ test: 'test' }} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('LoadingIndicator renders correctly', () => {
  const tree = renderer.create(<LoadingIndicator />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('LoadingVideo renders correctly', () => {
  const tree = renderer
    .create(<LoadingVideo source={'test'} style={{ test: 'test' }} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('Logo renders correctly', () => {
  const tree = renderer.create(<Logo />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('NetworkIndicator renders correctly', () => {
  const tree = renderer.create(<NetworkIndicator />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('SwiperImage renders correctly', () => {
  const tree = renderer.create(<SwiperImage media={['test']} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('UserHeader renders correctly', () => {
  const tree = renderer.create(<UserHeader username={'test'} />).toJSON();
  expect(tree).toMatchSnapshot();
});
