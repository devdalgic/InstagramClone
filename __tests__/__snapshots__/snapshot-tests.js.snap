// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Caption renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "marginStart": 10,
      "marginTop": 5,
    }
  }
>
  <Text
    style={
      Object {
        "color": "black",
      }
    }
  >
    <Text
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      test
       
    </Text>
    test
     
  </Text>
</View>
`;

exports[`FeedPost renders correctly 1`] = `
<View
  style={
    Object {
      "marginVertical": 15,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "marginBottom": 10,
        "marginStart": 10,
      }
    }
  >
    <MockedFontAwesome
      color="#841584"
      name="user-circle"
      size={32}
      style={
        Object {
          "borderRadius": 20,
          "marginRight": 10,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "black",
        }
      }
    >
      test
    </Text>
  </View>
  <Video
    onBuffer={[Function]}
    posterResizeMode="cover"
    repeat={true}
    resizeMode="cover"
    source={
      Object {
        "uri": "test",
      }
    }
    style={
      Object {
        "height": 300,
        "width": 750,
      }
    }
  />
  <ActivityIndicator
    animating={true}
    color="#841584"
    size="large"
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "marginStart": 10,
        "marginTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        hitSlop={
          Object {
            "bottom": 20,
            "left": 20,
            "right": 20,
            "top": 20,
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "marginHorizontal": 10,
            "opacity": 1,
          }
        }
      >
        <MockedFontAwesome
          color="black"
          name="heart-o"
          size={32}
        />
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        hitSlop={
          Object {
            "bottom": 20,
            "left": 20,
            "right": 20,
            "top": 20,
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "marginHorizontal": 10,
            "opacity": 1,
          }
        }
      >
        <MockedFontAwesome
          color="black"
          name="comment-o"
          size={32}
        />
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        hitSlop={
          Object {
            "bottom": 20,
            "left": 20,
            "right": 20,
            "top": 20,
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "marginHorizontal": 10,
            "opacity": 1,
          }
        }
      >
        <MockedFontAwesome
          color="black"
          name="paper-plane-o"
          size={32}
        />
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        hitSlop={
          Object {
            "bottom": 20,
            "left": 20,
            "right": 20,
            "top": 20,
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "position": "absolute",
            "right": 20,
          }
        }
      >
        <MockedFontAwesome
          color="black"
          name="bookmark-o"
          size={32}
        />
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "marginTop": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "black",
          }
        }
      >
        Liked by
        <Text
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
           
           
        </Text>
        and others
         
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "marginStart": 10,
        "marginTop": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "black",
        }
      }
    >
      <Text
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        test
         
      </Text>
      test
       
    </Text>
  </View>
</View>
`;

exports[`LikedBy renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "marginTop": 10,
    }
  }
>
  <Text
    style={
      Object {
        "color": "black",
      }
    }
  >
    Liked by
    <Text
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
       
      test
       
    </Text>
    and others
     
  </Text>
</View>
`;

exports[`LoadingImage renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Image
    onLoadEnd={[Function]}
    source={
      Object {
        "uri": "test",
      }
    }
    style={
      Object {
        "test": "test",
      }
    }
  />
  <ActivityIndicator
    animating={true}
    color="#841584"
    size="large"
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
</View>
`;

exports[`LoadingIndicator renders correctly 1`] = `
<ActivityIndicator
  animating={true}
  color="#841584"
  size="large"
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
/>
`;

exports[`LoadingVideo renders correctly 1`] = `
Array [
  <Video
    onBuffer={[Function]}
    posterResizeMode="cover"
    repeat={true}
    resizeMode="cover"
    source={
      Object {
        "uri": "test",
      }
    }
    style={
      Object {
        "test": "test",
      }
    }
  />,
  <ActivityIndicator
    animating={true}
    color="#841584"
    size="large"
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />,
]
`;

exports[`Logo renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <MockedFontAwesome5
    color="#841584"
    name="cat"
    size={192}
    style={
      Object {
        "marginVertical": 10,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "black",
      }
    }
  >
    Case Study by Denizhan Dalgic
  </Text>
  <Text
    style={
      Object {
        "color": "black",
      }
    }
  >
    Instagram Clone
  </Text>
</View>
`;

exports[`NetworkIndicator renders correctly 1`] = `<View />`;

exports[`SwiperImage renders correctly 1`] = `
<RCTScrollView
  data={
    Array [
      "test",
    ]
  }
  decelerationRate="normal"
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={true}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  showsHorizontalScrollIndicator={false}
  snapToAlignment="start"
  snapToInterval={750}
  stickyHeaderIndices={Array []}
  viewabilityConfigCallbackPairs={Array []}
>
  <View>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Image
          onLoadEnd={[Function]}
          source={
            Object {
              "uri": "test",
            }
          }
          style={
            Object {
              "height": 300,
              "width": 500,
            }
          }
        />
        <ActivityIndicator
          animating={true}
          color="#841584"
          size="large"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`UserHeader renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "marginBottom": 10,
      "marginStart": 10,
    }
  }
>
  <MockedFontAwesome
    color="#841584"
    name="user-circle"
    size={32}
    style={
      Object {
        "borderRadius": 20,
        "marginRight": 10,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "black",
      }
    }
  >
    test
  </Text>
</View>
`;
